cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -555,6 +557,38 @@ impl Environment for ManagedEnvironment {
 
 /// Constructors and related functions
 impl ManagedEnvironment {
+    /// If there's an out of sync local checkout, ensure it's locked.
+    /// If the checkout is in sync, return it's lock contents.
+    ///
+    /// This errors if an in-sync checkout doesn't have a lockfile, since that's
+    /// a bad state.
+    fn ensure_locked(
+        &mut self,
+        flox: &Flox,
+        local_checkout: &mut CoreEnvironment,
+    ) -> Result<LockedManifest, EnvironmentError> {
+        // Otherwise, there would be a generation without a lockfile, which is a bad state,
+        // and we error.
+        if !Self::validate_checkout(local_checkout, &self.get_current_generation(flox)?)? {
+            Ok(local_checkout.ensure_locked(flox)?)
+        } else {
+            let content = local_checkout.existing_deserialized_lockfile()?;
+            content.ok_or(EnvironmentError::MissingLockfile)
+        }
+    }
+
+    pub fn build(&mut self, flox: &Flox) -> Result<PathBuf, EnvironmentError> {
+        let mut local_checkout = self.local_env_or_copy_current_generation(flox)?;
+
+        Ok(local_checkout.build(flox)?)
+    }
```

Comment:
question: why does this `build()` impl not link anymore?

Reply:
I think it's a bit more consistent with the approach we have in core environments after 4d3b19e7fda0f439d164353a9c1a48ed55a0025b. It would be inconsistent for build to sometimes mean build and sometimes mean build && link.

In general I get the sense keeping lock + build + link separate will cause us less pain in the future.

But currently everywhere this is called we do a link, so we don't necessarily have to split it if you feel strongly

Reply:
I'm happy to split the steps